controllers/k8ssandra/stargate.go

```diff
@@ -192,3 +198,104 @@ func (r *K8ssandraClusterReconciler) removeStargateStatus(kc *api.K8ssandraClust
 		}
 	}
 }
+
+func (r *K8ssandraClusterReconciler) reconcileStargateConfigMap(
+	ctx context.Context,
+	remoteClient client.Client,
+	kc *api.K8ssandraCluster,
+	desiredConfig map[string]interface{},
+	userConfigMapContent,
+	namespace string,
+	logger logr.Logger,
+) result.ReconcileResult {
+	logger.Info("Reconciling Stargate Cassandra yaml configMap on namespace : " + namespace)
+	filteredCassandraConfig := filterYamlConfig(desiredConfig["cassandra-yaml"].(map[string]interface{}))
+
+	configYamlString := ""
+	if len(filteredCassandraConfig) > 0 {
+		if configYaml, err := yaml.Marshal(filteredCassandraConfig); err != nil {
+			return result.Error(err)
+		} else {
+			configYamlString = string(configYaml)
+		}
+	}
+
+	if userConfigMapContent != "" {
+		separator := "\n"
+		if strings.HasSuffix(userConfigMapContent, "\n") {
+			separator = ""
+		}
+		configYamlString = userConfigMapContent + separator + configYamlString
+	}
+
+	configMapKey := client.ObjectKey{
+		Namespace: namespace,
+		Name:      stargate.CassandraConfigMap,
+	}
+
+	logger = logger.WithValues("StargateConfigMap", configMapKey)
+	desiredConfigMap := createStargateConfigMap(namespace, configYamlString, kc)
+	// Compute a hash which will allow to compare desired and actual configMaps
+	annotations.AddHashAnnotation(desiredConfigMap)
+	actualConfigMap := &corev1.ConfigMap{}
+
+	if err := remoteClient.Get(ctx, configMapKey, actualConfigMap); err != nil {
+		if errors.IsNotFound(err) {
+			logger.Info("Stargate configMap doesn't exist, creating it")
+			if err := remoteClient.Create(ctx, desiredConfigMap); err != nil {
```

Comment:
You should set an owner reference on the ConfigMap created by the controller so that deletes cascade.